### PR TITLE
feat(mdx): Add support for turning ![]() into <Image>

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

With this PR, we now have optimized images in Markdown, MDX and Markdoc! Wow!

Fix https://github.com/withastro/astro/issues/6777

## Testing

Added a test

## Docs

https://github.com/withastro/docs/pull/3032

### CODE REVIEW

File: packages/integrations/mdx/src/remark-images-to-component.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+				parent!.children.splice(index!, 1, componentElement);
+			}
+		});
+
+		// Add all the import statements to the top of the file for the images
+		tree.children.unshift(...importsStatements);
+
+		// Add an import statement for the Astro Image component, we rename it to avoid conflicts
+		tree.children.unshift(jsToTreeNode(`import { Image as __AstroImage__ } from "astro:assets";`));
```
Comment (jasikpark) [inline packages/integrations/mdx/src/remark-images-to-component.ts:96]: Is it possible to import the same binding twice from a module if they have different names?

**Review by jasikpark — COMMENTED** [review]
this looks great, i'm excited!

File: packages/integrations/mdx/test/fixtures/mdx-images/src/pages/index.mdx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+![Houston](../assets/houston.webp)
+
```
Comment (ematipico) [inline packages/integrations/mdx/test/fixtures/mdx-images/src/pages/index.mdx:2]: Can we add the following test cases:
- an image with "title"
- test case where we correctly rename the image? Ideally, we should hit the code where we use `replace(/\W/g, '_')`.

File: packages/integrations/mdx/src/remark-images-to-component.ts (hunk 1 above)
Comment (Princesseuh) [reply packages/integrations/mdx/src/remark-images-to-component.ts:96]: Yes! As long as you rename it it's ok

File: packages/integrations/mdx/test/fixtures/mdx-images/src/pages/index.mdx (hunk 2 above)
Comment (Princesseuh) [reply packages/integrations/mdx/test/fixtures/mdx-images/src/pages/index.mdx:2]: Will do, thank you for the callout!

**Review by ematipico — APPROVED** [review]

**Review by bluwy — APPROVED** [review]

**Pull request merged**